### PR TITLE
Refactor before connecting MessageQueue to MessageHandler

### PR DESCRIPTION
**Pull request by xitronix**

### CODE REVIEW

File: universal-login-relayer/lib/relayer.ts [hunk 1]
```diff
@@ -70,8 +68,7 @@ class Relayer {
     this.walletContractService = new WalletService(this.wallet, this.config.walletMasterAddress, this.ensService, this.hooks);
     this.pendingMessagesStore = new PendingMessagesSQLStore(this.database);
     this.transactionQueueStore = new TransactionQueueStore(this.database);
-    this.transactionQueueService = new TransactionQueueService(this.wallet, this.provider, this.transactionQueueStore);
-    this.messageHandler = new MessageHandler(this.wallet, this.authorisationService, this.hooks, this.transactionQueueService, this.pendingMessagesStore, this.config.contractWhiteList);
+    this.messageHandler = new MessageHandler(this.wallet, this.authorisationService, this.hooks, this.pendingMessagesStore, this.transactionQueueStore, this.config.contractWhiteList, this.provider);
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/relayer.ts]: `provider` is already in the wallet

File: universal-login-relayer/lib/services/MessageHandler.ts [hunk 2]
```diff
@@ -9,19 +9,22 @@ import {decodeDataForExecuteSigned} from './transactions/serialisation';
...
+import ITransactionQueueStore from './transactions/ITransactionQueueStore';
 
 class MessageHandler {
   private pendingMessages: PendingMessages;
+  private messageQueue: TransactionQueueService;
   private executor: MessageExecutor;
   private validator: MessageValidator;
 
-  constructor(private wallet: Wallet, private authorisationService: AuthorisationService, private hooks: EventEmitter, private transactionQueue: TransactionQueueService, pendingMessagesStore: IPendingMessagesStore, contractWhiteList: ContractWhiteList) {
+  constructor(private wallet: Wallet, private authorisationService: AuthorisationService, private hooks: EventEmitter, pendingMessagesStore: IPendingMessagesStore, transactionQueueStore: ITransactionQueueStore, contractWhiteList: ContractWhiteList, provider: providers.Provider) {
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/services/MessageHandler.ts]: `provider` is already in the `wallet`

File: universal-login-relayer/lib/services/MessageHandler.ts [hunk 3]
```diff
@@ -9,19 +9,22 @@ import {decodeDataForExecuteSigned} from './transactions/serialisation';
...
     this.validator = new MessageValidator(this.wallet, contractWhiteList);
     this.executor = new MessageExecutor(
       this.wallet,
       this.onTransactionSent.bind(this),
       this.validator
       );
+    this.messageQueue = new TransactionQueueService(this.wallet, provider, transactionQueueStore);
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/services/MessageHandler.ts]: `provider` is already in the `wallet`

File: universal-login-relayer/test/config/message.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import {TEST_ACCOUNT_ADDRESS, createSignedMessage} from '@universal-login/commons';
+
+
+export const testPrivateKey = '0x63f01680950dc70f2eb8f373de0c360fcbb89ef437f2f6f2f0a1797979e490a4';
+export const getSignedMessage = async () => createSignedMessage({from: TEST_ACCOUNT_ADDRESS, to: TEST_ACCOUNT_ADDRESS}, testPrivateKey);
```
Comment (marekkirejczyk) [inline universal-login-relayer/test/config/message.ts]: getTestSignedMessage

File: universal-login-relayer/test/relayer/services/transactions/TransactionQueueStore.ts [hunk 5]
```diff
@@ -31,7 +31,7 @@ describe('INT: Transaction Queue Store', async () => {
 
   it('transaction round trip', async () => {
     const [idFirst] = await transactionQueueStore.add(signedMessage);
-    const signedMessage2 = await await createSignedMessage({from: TEST_ACCOUNT_ADDRESS, to: TEST_ACCOUNT_ADDRESS, value: utils.parseEther('2')}, privateKey);
+    const signedMessage2 = await createSignedMessage({from: TEST_ACCOUNT_ADDRESS, to: TEST_ACCOUNT_ADDRESS, value: utils.parseEther('2')}, testPrivateKey);
```
Comment (marekkirejczyk) [inline universal-login-relayer/test/relayer/services/transactions/TransactionQueueStore.ts]: `getTestSignedMessage({value: utils.parseEther('2')})`

File: universal-login-relayer/lib/services/transactions/ITransactionQueueStore.ts [hunk 6]
```diff
@@ -1,17 +1,17 @@
-import {utils} from 'ethers';
+import {SignedMessage} from '@universal-login/commons';
 
 export interface ITransactionQueueStore {
-  add: (transaction: Partial<utils.Transaction>) => Promise<string>;
-  getNext: () => Promise<TransactionEntity | undefined>;
+  add: (transaction: SignedMessage) => Promise<string>;
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/services/transactions/ITransactionQueueStore.ts]: 👍

File: universal-login-relayer/lib/services/transactions/TransactionQueueMemoryStore.ts [hunk 7]
```diff
@@ -1,19 +1,19 @@
-import {utils} from 'ethers';
-import {ITransactionQueueStore, TransactionEntity} from './ITransactionQueueStore';
+import {SignedMessage} from '@universal-login/commons';
+import {ITransactionQueueStore, MessageEntity} from './ITransactionQueueStore';
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/services/transactions/TransactionQueueMemoryStore.ts]: We should probably move this file to `test` folder, as it should not be used outside of tests.

File: universal-login-relayer/test/relayer/services/transactions/TransactionQueueStore.ts [hunk 8]
```diff
@@ -2,15 +2,16 @@ import {expect} from 'chai';
...
-const transaction2 = {
-  ...transaction,
-  value: utils.parseEther('2')
-};
+import {SignedMessage, createSignedMessage, TEST_ACCOUNT_ADDRESS} from '@universal-login/commons';
 
 describe('INT: Transaction Queue Store', async () => {
   let transactionQueueStore: TransactionQueueStore;
+  let signedMessage: SignedMessage;
+  const privateKey = '0x63f01680950dc70f2eb8f373de0c360fcbb89ef437f2f6f2f0a1797979e490a4';
```
Comment (marekkirejczyk) [inline universal-login-relayer/test/relayer/services/transactions/TransactionQueueStore.ts]: Why not introduce `TEST_PRIVATE_KEY`?

File: universal-login-relayer/lib/relayer.ts (hunk 1 above)
Comment (xitronix) [reply universal-login-relayer/lib/relayer.ts]: ✔️

File: universal-login-relayer/test/config/message.ts (hunk 4 above)
Comment (xitronix) [reply universal-login-relayer/test/config/message.ts]: ✔️

File: universal-login-relayer/test/relayer/services/transactions/TransactionQueueStore.ts (hunk 5 above)
Comment (xitronix) [reply universal-login-relayer/test/relayer/services/transactions/TransactionQueueStore.ts]: ✔️

File: universal-login-relayer/test/relayer/services/transactions/TransactionQueueStore.ts (hunk 8 above)
Comment (xitronix) [reply universal-login-relayer/test/relayer/services/transactions/TransactionQueueStore.ts]: I think it is not actual already.

File: universal-login-relayer/lib/services/MessageHandler.ts (hunk 2 above)
Comment (xitronix) [reply universal-login-relayer/lib/services/MessageHandler.ts]: ✔️

**Pull request merged**